Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/ExposureSummaryRepository.kt

```diff
@@ -1,70 +1,30 @@
 package de.rki.coronawarnapp.storage
 
-import com.google.android.gms.nearby.exposurenotification.ExposureSummary
-import de.rki.coronawarnapp.CoronaWarnApplication
-import de.rki.coronawarnapp.nearby.InternalExposureNotificationClient
+import com.google.android.gms.nearby.exposurenotification.ExposureWindow
+import de.rki.coronawarnapp.risk.RiskLevels
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
+import javax.inject.Inject
 
-class ExposureSummaryRepository(private val exposureSummaryDao: ExposureSummaryDao) {
-    companion object {
-        @Volatile
-        private var instance: ExposureSummaryRepository? = null
-
-        private fun getInstance(exposureSummaryDao: ExposureSummaryDao) =
-            instance ?: synchronized(this) {
-                instance ?: ExposureSummaryRepository(exposureSummaryDao).also { instance = it }
-            }
-
-        fun resetInstance() = synchronized(this) {
-            instance = null
-        }
-
-        fun getExposureSummaryRepository(): ExposureSummaryRepository {
-            return getInstance(
-                AppDatabase.getInstance(CoronaWarnApplication.getAppContext())
-                    .exposureSummaryDao()
-            )
-        }
+class ExposureSummaryRepository @Inject constructor(
+    private val riskLevels: RiskLevels
+) {
 
+    companion object {
         private val internalMatchedKeyCount = MutableStateFlow(0)
         val matchedKeyCount: Flow<Int> = internalMatchedKeyCount
 
         private val internalDaysSinceLastExposure = MutableStateFlow(0)
         val daysSinceLastExposure: Flow<Int> = internalDaysSinceLastExposure
     }
 
-    suspend fun getExposureSummaryEntities() = exposureSummaryDao.getExposureSummaryEntities()
-        .map { it.convertToExposureSummary() }
+    private var entities = emptyList<ExposureWindow>()
```

Comment:
This was for the debug UIs right? We could make this a `MutableStateFlow` and then the UI could automatically update on new data.

Reply:
will change it